image_transport/test/test_publisher.cpp

```diff
@@ -46,31 +46,87 @@ class TestPublisher : public ::testing::Test
   rclcpp::Node::SharedPtr node_;
 };
 
-TEST_F(TestPublisher, Publisher) {
+TEST_F(TestPublisher, publisher) {
   auto pub = image_transport::create_publisher(node_.get(), "camera/image");
 }
 
-TEST_F(TestPublisher, ImageTransportPublisher) {
+TEST_F(TestPublisher, image_transport_publisher) {
   image_transport::ImageTransport it(node_);
   auto pub = it.advertise("camera/image", 1);
 }
 
-TEST_F(TestPublisher, CameraPublisher) {
+TEST_F(TestPublisher, camera_publisher) {
   auto camera_pub = image_transport::create_camera_publisher(node_.get(), "camera/image");
 }
 
-TEST_F(TestPublisher, ImageTransportCameraPublisher) {
+TEST_F(TestPublisher, image_transport_camera_publisher) {
   image_transport::ImageTransport it(node_);
   auto pub = it.advertiseCamera("camera/image", 1);
 }
 
-TEST_F(TestPublisher, Shutdown) {
+TEST_F(TestPublisher, shutdown) {
   auto pub = image_transport::create_publisher(node_.get(), "camera/image");
   EXPECT_EQ(node_->get_node_graph_interface()->count_publishers("camera/image"), 1u);
   pub.shutdown();
   EXPECT_EQ(node_->get_node_graph_interface()->count_publishers("camera/image"), 0u);
 }
 
+TEST_F(TestPublisher, qos_override) {
+  // without options
+  auto pub =
+    image_transport::create_publisher(node_.get(), "camera/image", rmw_qos_profile_default);
+  auto endpoint_info_vec = node_->get_publishers_info_by_topic("camera/image");
+  EXPECT_EQ(endpoint_info_vec[0].qos_profile().reliability(), rclcpp::ReliabilityPolicy::Reliable);
+  pub.shutdown();
+
+  node_ = rclcpp::Node::make_shared(
+    "test_publisher", rclcpp::NodeOptions().parameter_overrides(
+  {
+    rclcpp::Parameter(
+      "qos_overrides./camera/image.publisher.reliability", "best_effort"),
+  }));
+  pub = image_transport::create_publisher(node_.get(), "camera/image", rmw_qos_profile_default);
+
+  endpoint_info_vec = node_->get_publishers_info_by_topic("camera/image");
+  EXPECT_EQ(
+    endpoint_info_vec[0].qos_profile().reliability(),
+    rclcpp::ReliabilityPolicy::Reliable);
+  pub.shutdown();
+
+  // with options
+  node_ = rclcpp::Node::make_shared("test_publisher");
```

Comment:
You're making an entire node here just to replace it later on on line 114. Line 114 should create a new node with a new variable name rather than overwriting the one from the test fixture. You could even consider making this a stand-alone test case rather than using the test fixture, since you don't actually use the node created by the test fixture.

Reply:
separate qos override test case at https://github.com/ros-perception/image_common/pull/208/commits/fb89dede67ea2fae6d4903f96272e34461fff195